buf/registry/plugin/v1beta1/plugin_service.proto

```diff
@@ -0,0 +1,217 @@
+    option idempotency_level = IDEMPOTENT;
+  }
+  // Update existing Plugins.
+  //
+  // This operation is atomic. Either all Plugins are updated or an error is returned.
+  rpc UpdatePlugins(UpdatePluginsRequest) returns (UpdatePluginsResponse) {
+    option idempotency_level = IDEMPOTENT;
+  }
+  // Delete existing Plugins.
+  //
+  // This operation is atomic. Either all Plugins are deleted or an error is returned.
+  rpc DeletePlugins(DeletePluginsRequest) returns (DeletePluginsResponse) {
+    option idempotency_level = IDEMPOTENT;
+  }
+}
+
+message GetPluginsRequest {
+  // The Plugins to request.
+  repeated PluginRef plugin_refs = 1 [
+    (buf.validate.field).repeated.min_items = 1,
+    (buf.validate.field).repeated.max_items = 250
+  ];
+}
+
+message GetPluginsResponse {
+  // The retrieved Plugins in the same order as requested.
+  repeated Plugin plugins = 1 [(buf.validate.field).repeated.min_items = 1];
+}
+
+message ListPluginsRequest {
+  // The list order.
+  enum Order {
+    ORDER_UNSPECIFIED = 0;
+    // Order by create_time newest to oldest.
+    ORDER_CREATE_TIME_DESC = 1;
+    // Order by create_time oldest to newest.
+    ORDER_CREATE_TIME_ASC = 2;
+  }
+  // Filter for the list request, if any. When multiple filters are specified, they are combined.
+  message Filter {
+    // The specific Organizations to list Plugins for.
+    //
+    // If empty, all Plugins for all Organizations are listed, but this functionality is limited to
+    // Users with the necessary permissions.
+    repeated buf.registry.owner.v1.OwnerRef owner_refs = 1;
+    // The collections to list Plugins for.
+    //
+    // If empty, all Plugins are listed regardless of collection.
+    repeated string collection_names = 2;
+    // The languages to list Plugins for.
+    //
+    // If empty, all Plugins are listed regardless of language.
+    repeated PluginLanguage languages = 3 [
+      (buf.validate.field).repeated.unique = true,
+      (buf.validate.field).repeated.items.enum.defined_only = true,
+      (buf.validate.field).repeated.items.enum.not_in = 0
+    ];
+    // The registry types to list Plugins for.
+    //
+    // If empty, all Plugins are listed regardless of registry type.
+    repeated PluginRegistryType registry_types = 4 [
+      (buf.validate.field).repeated.unique = true,
+      (buf.validate.field).repeated.items.enum.defined_only = true,
+      (buf.validate.field).repeated.items.enum.not_in = 0
+    ];
+  }
+  // The maximum number of items to return.
+  //
+  // The default value is 10.
+  uint32 page_size = 1 [(buf.validate.field).uint32.lte = 250];
+  // The page to start from.
+  //
+  // If empty, the first page is returned.
+  string page_token = 2 [(buf.validate.field).string.max_len = 4096];
+  // An optional filter to filter the list of Plugins. If unset, all Plugins for all organizations
+  // are listed, but this functionality is limited to Users with the necessary permissions.
+  Filter filter = 3;
```

Comment:
Why was this put under a nested message, when it was not in the module API? Ie what is the reasoning used here to add inconsistency to our APIs across packages?

Reply:
I think having a `Filter` message aids in readability, especially as the number of items you filter by increases.

Having a single nested message to hold all the optional filters felt more appropriate instead of mixing it with other fields like page_size, page_token and order (sorting)

Maybe we could update modules to have a message `Filter` and make this a consistent pattern?

Reply:
I think the conclusion I take here is that there isn't a strong reason to add inconsistency - readability is a judgement call. Judgement calls in style take a back seat to consistency in APIs, and the non-nested-message version has already shipped in v1, therefore please match the existing pattern.

Reply:
Done. d9ba013e